netty/src/main/java/io/grpc/netty/GrpcHttp2HeadersUtils.java

```diff
@@ -340,7 +340,12 @@ public Http2Headers add(CharSequence csName, CharSequence csValue) {
       AsciiString name = validateName(requireAsciiString(csName));
       AsciiString value = requireAsciiString(csValue);
       if (isPseudoHeader(name)) {
-        addPseudoHeader(name, value);
+        AsciiString previous = getPseudoHeader(name);
```

Comment:
It'd be possible to implement this without a get-then-set if `setPseudoHeader()` returned the old value. We'd have to re-set the old value if there was one already present, but that's not the worst. Returning-old-value`setPseudoHeader()` would seem to most benefit `remove()`, but it isn't safe for remove because it would throw for `remove(":status')`. Having the returning-old-value `setPseudoHeader()` made it really easy to make mistakes, so I went with the KISS approach. These code paths don't matter for performance.